document-generator-prosecution/src/test/java/uk/gov/companieshouse/document/generator/prosecution/ProsecutionDocumentInfoServiceTest.java

```diff
@@ -1,101 +1,91 @@
-                    "/prosecution/ultimatum" + PROSECUTION_CASE_URI;
-    private static final String SJP_RESOURCE_URI = "/prosecution/sjp" + PROSECUTION_CASE_URI;
 
-    @Mock
-    private UltimatumDocumentInfoBuilderProvider docInfoBuilderProvider;
-
-    @Mock
-    private UltimatumDocumentInfoBuilder docInfoBuilder;
+    @InjectMocks
+    ProsecutionDocumentInfoService prosecutionDocumentInfoService;
 
     @Mock
-    ProsecutionClient prosecutionClient;
+    ProsecutionHandler mockHandler;
 
-    @Mock
-    ProsecutionCase prosecutionCase;
+    private String resourceUri = "uri";
+    private String requestId = "1";
 
-    @DisplayName("Given a request to produce an ultimatum for a given prosecution case, "
-                    + "when the UltimatumDocumentInfoService successfully produces a DocumentInfoResponse,"
-                    + "then that response is returned.")
     @Test
-    void testGetDocumentInfoForUltimatum()
-                    throws DocumentInfoException, SdkException, DocumentInfoCreationException {
-        ProsecutionDocumentInfoService prosDocInfoService = new ProsecutionDocumentInfoService(
-                        docInfoBuilderProvider, prosecutionClient);
-        when(prosecutionClient.getProsecutionCase(PROSECUTION_CASE_URI, REQUEST_ID))
-                        .thenReturn(prosecutionCase);
-        when(docInfoBuilderProvider.builder()).thenReturn(docInfoBuilder);
-        DocumentInfoResponse responseToReturn = new DocumentInfoResponse();
-        when(docInfoBuilder.build()).thenReturn(responseToReturn);
-
-        DocumentInfoRequest docInfoReq = new DocumentInfoRequest();
-        docInfoReq.setResourceUri(ULTIMATUM_RESOURCE_URI);
-        docInfoReq.setRequestId(REQUEST_ID);
-        DocumentInfoResponse result = prosDocInfoService.getDocumentInfo(docInfoReq);
-        Assertions.assertSame(responseToReturn, result);
+    @DisplayName("Test using a prosecution case that has an ACCEPTED status should call "
+            + "the handler for ultimatum")
+    public void testGetDocumentInfoWithAcceptedCase() throws Exception{
+        ProsecutionCase prosecutionCase = new ProsecutionCase();
+        prosecutionCase.setStatus(ProsecutionCaseStatusApi.ACCEPTED);
+        DocumentInfoRequest request = getDocumentInfoRequest(resourceUri, requestId);
+        ProsecutionDocument document = new ProsecutionDocument();
+        document.setProsecutionCase(prosecutionCase);
+        when(mockHandler.getProsecutionDocument(resourceUri)).thenReturn(document);
+        DocumentInfoResponse response = new DocumentInfoResponse();
+        when(mockHandler.getDocumentResponse(document, requestId, prosecutionCase.getStatus())).thenReturn(response);
+        prosecutionDocumentInfoService.getDocumentInfo(request);
+        verify(mockHandler, times(1)).getProsecutionDocument(resourceUri);
+        verify(mockHandler, times(1)).getDocumentResponse(document, requestId, prosecutionCase.getStatus());
```

Comment:
Please add line spacing to improve the readability of this

Reply:
This test has been refactored along with the line spacing.